### ISSUE TITLE
SC version number at command prompt

### ISSUE THREAD

**Issue by malx122** [Starcounter/Starcounter#2545]
Would be very nice if you can get this to work!
only shows 
```
>star --version
version=
```
I need it in order to be able to compile in which version of starcounter our application has been built against. It will help alot for avoiding confusing between our developers.

**Comment by per-samuelsson** [comment]
Hi @malx122,

This actually worked once, when I first implemented it. And then stopped working for some reason. I agree it would be nice to have.

I'll assign it to me to do eventually.

**Comment by malx122** [comment]
I would prefer more than eventually... :)
(We have problem all the time with knowing which version we have built apps against)

**Comment by per-samuelsson** [comment]
> I would prefer more than eventually

OK, I'll bring it up with @dan31 to see how I should prioritize. I'll get back with an answer at the latest tomorrow.

**Comment by dan31** [comment]
@malx122, we are getting into the problem now.

**Comment by andwah** [comment]
You can see the current starcounter version in the Administrator

[http://localhost:8181/#/serverSettings](http://localhost:8181/#/serverSettings)

![image](https://cloud.githubusercontent.com/assets/2656996/6104221/cce53874-b04b-11e4-97fa-7a20c89a4845.png)

**Comment by parherman** [comment]
I added a test for this in commit 5fceafff722fcb9d4c094519c3bfce5ebe679ee3

**Comment by malx122** [comment]
@andwah I know, but does not help if you want to compile into your application which version you have built the application against.

**Comment by andwah** [comment]
in your app you can access this property
 ```
Starcounter.Internal.CurrentVersion.Version
```
will this work for you?

**Comment by malx122** [comment]
@andwah Nope, I want the version the app is built against, not the one its running on.

**Comment by per-samuelsson** [comment]
Just a note, that me and @parherman discovered: ```star --info``` also does not provide any version. This should also be fixed.

**Comment by per-samuelsson** [comment]
> @andwah Nope, I want the version the app is built against, not the one its running on.

This is not currently possible (not easily anyway). The version info shown by ```star -version``` is the current installation version.

**Comment by per-samuelsson** [comment]
There is a (rather old) issue that request the same thing you are after, @malx122: #525. This is not implemented though (see prev comment).

**Comment by malx122** [comment]
@per-samuelsson It is very easy when star -version works. I have a prebuild cmd file that runs different cmds and then stored them in an textfile that is "built in". Thats how i get git data for the current build as well. (I can see exactly which code that has been built and also if its "dirty" or not)

**Comment by per-samuelsson** [comment]
> It is very easy when star -version works.

I see. I mainly wanted to point out that ```star -v``` does not show the version of any particular application; it shows the version of Starcounter installed. So if you compile ```foo.exe``` against Starcounter 1.2.3, reinstall Starcounter from 1.2.3 -> 4.5.6, doing ```star --version``` will give you 4.5.6.

With #525, we have plans to "tag" any application with the version it was build aginst.

**Comment by malx122** [comment]
Yea, but having it stored in a prebuild created textfile compiled in as a resource will of course be static after build. It actually a very good way to store build info. (Like I said, git info, build computer name e.t.c.).

**Comment by parherman** [comment]
This issue was fixed in commit 03e873b3d71d2c617a3c62054274e19f8ae5f7a3
